src/main/java/bio/terra/service/filedata/flight/ingest/IngestFileGetProjectStep.java

```diff
@@ -0,0 +1,47 @@
+package bio.terra.service.filedata.flight.ingest;
+
+import bio.terra.model.BillingProfileModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.filedata.flight.FileMapKeys;
+import bio.terra.service.profile.flight.ProfileMapKeys;
+import bio.terra.service.resourcemanagement.DataLocationSelector;
+import bio.terra.service.resourcemanagement.ResourceService;
+import bio.terra.service.resourcemanagement.google.GoogleProjectResource;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+
+public class IngestFileGetProjectStep implements Step {
+  private final ResourceService resourceService;
+  private final Dataset dataset;
+  private final DataLocationSelector dataLocationSelector;
+
+  public IngestFileGetProjectStep(
+      ResourceService resourceService, Dataset dataset, DataLocationSelector dataLocationSelector) {
+    this.resourceService = resourceService;
+    this.dataset = dataset;
+    this.dataLocationSelector = dataLocationSelector;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) {
+    // Requests a google project from RBS and puts it in the working map
```

Comment:
Man, I with steps had a `getDescription` method.  It would be great for logging and monitoring.  Short of that, would it make sense to move this to javadoc on the class?

Reply:
described